ios/web_view/internal/cwv_web_view_extras.mm

```diff
@@ -0,0 +1,127 @@
+  // This runs `UpdateScripts` on the configuration provider which we will need
+  // to call in-place of `-[WKUserContentController removeAllUserScripts]` until
+  // all Brave JavaScript features are ported over to actual Chromium
+  // JavascriptFeature types and added to BraveWebClient
+  web::WKWebViewConfigurationProvider& config_provider =
+      web::WKWebViewConfigurationProvider::FromBrowserState(
+          self.webState->GetBrowserState());
+  config_provider.UpdateScripts();
+}
+
+- (void)createPDF:(void (^)(NSData* _Nullable))completionHandler {
+  self.webState->CreateFullPagePdf(base::BindOnce(completionHandler));
+}
+
+- (void)takeSnapshotWithRect:(CGRect)rect
+           completionHandler:(void (^)(UIImage* _Nullable))completionHandler {
+  self.webState->TakeSnapshot(rect, base::BindRepeating(completionHandler));
+}
+
+- (CWVUserAgentType)currentItemUserAgentType {
+  return static_cast<CWVUserAgentType>(self.webState->GetNavigationManager()
+                                           ->GetVisibleItem()
+                                           ->GetUserAgentType());
+}
+
+- (void)reloadWithUserAgentType:(CWVUserAgentType)userAgentType {
+  self.webState->GetNavigationManager()->ReloadWithUserAgentType(
+      static_cast<web::UserAgentType>(userAgentType));
+}
+
+- (void)evaluateJavaScript:(NSString*)javaScriptString
+              contentWorld:(WKContentWorld*)contentWorld
+         completionHandler:(void (^)(id result, NSError* error))completion {
+  web::WebFrame* mainFrame =
+      self.webState->GetPageWorldWebFramesManager()->GetMainWebFrame();
+  if (!mainFrame) {
+    if (completion) {
+      completion(nil, [NSError errorWithDomain:@"org.chromium.chromewebview"
+                                          code:0
+                                      userInfo:nil]);
+    }
+    return;
+  }
+  web::WebFrameInternal* webFrame = mainFrame->GetWebFrameInternal();
+  auto worlds = web::JavaScriptFeatureManager::FromBrowserState(
+                    self.configuration.browserState)
+                    ->GetAllContentWorlds();
+
+  auto jsContentWorld =
+      std::find_if(worlds.begin(), worlds.end(), [&contentWorld](auto world) {
+        return world->GetWKContentWorld() == contentWorld;
+      });
+  if (jsContentWorld == worlds.end()) {
+    if (completion) {
+      completion(nil, [NSError errorWithDomain:@"org.chromium.chromewebview"
+                                          code:0
+                                      userInfo:nil]);
+    }
+    return;
+  }
+
+  webFrame->ExecuteJavaScriptInContentWorld(
+      base::SysNSStringToUTF16(javaScriptString), *jsContentWorld,
+      base::BindOnce(^(const base::Value* result, NSError* error) {
+        if (completion) {
+          id jsResult = nil;
+          if (!error && result) {
+            jsResult = [CWVWebView objectFromValue:result];
+          }
+          completion(jsResult, error);
+        }
+      }));
+}
+
+- (WKWebView*)underlyingWebView {
```

Comment:
why aren't we just exposing the `CRWWebController` as an attribute and then we can also use that for `createPDF`, etc...

Reply:
As per DM:
- We don't want to expose `CRWWebController` as its essentially the `WKWebView` handler, which we want to avoid using and overtime will use it less (for when we eventually can support a `use_blink` build)
- We can't expose `CRWWebController` as-is, would have to wrap it anyways

Reply:
discussed in DM, but maybe this should be `internalWebView` or something?